terraform/lib/dependabot/terraform/file_parser.rb

```diff
@@ -257,6 +257,35 @@ def version_from_ref(ref)
         ref.match(version_regex).named_captures.fetch("version")
       end
 
+      # rubocop:disable Metrics/PerceivedComplexity
+      # See https://www.terraform.io/docs/modules/sources.html#http-urls for
+      # details of how Terraform handle HTTP(S) sources for modules
+      def get_proxied_source(raw_source) # rubocop:disable Metrics/AbcSize
```

Comment:
The Terraform::RegistryClient actually already [has a copy of #get_proxied_source](https://github.com/dependabot/dependabot-core/blob/994075e9975e399fa0a7384b4e6ece9b872ae501/terraform/lib/dependabot/terraform/registry_client.rb#L27) in `main`. 
It's being used in the FileParser on [line 171](https://github.com/dependabot/dependabot-core/pull/5407/files#diff-ae8ba34c4ac15d002de6506c383c15096471b36b1e5642f8dfd6ce1825ba06b1R171)

It was moved from the Terraform::FileParser after this PR was created by https://github.com/dependabot/dependabot-core/pull/5366/ 

So we can either move this method back into the FileParser, undoing the changes in #5366, or we can remove it from this PR

Reply:
Addressed in [c2c188f](https://github.com/dependabot/dependabot-core/pull/5407/commits/c2c188ffdf2768d7ace7fc3f64b2263eddd94d0f)